staticaddr/manager.go

```diff
@@ -0,0 +1,167 @@
+package staticaddr
+
+import (
+	"context"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop"
+	"github.com/lightninglabs/loop/staticaddr/script"
+	staticaddressrpc "github.com/lightninglabs/loop/swapserverrpc"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+// ManagerConfig holds the configuration for the address manager.
+type ManagerConfig struct {
+	AddressClient staticaddressrpc.StaticAddressServerClient
+
+	LoopClient *loop.Client
+
+	Store AddressStore
+
+	Lnd lndclient.LndServices
+}
+
+// AddressManager manages the address state machines.
+type AddressManager struct {
+	cfg *ManagerConfig
+
+	runCtx context.Context
+
+	initChan chan struct{}
+
+	sync.Mutex
+}
+
+// NewAddressManager creates a new address manager.
+func NewAddressManager(cfg *ManagerConfig) *AddressManager {
+	return &AddressManager{
+		cfg:      cfg,
+		initChan: make(chan struct{}),
+	}
+}
+
+// Run runs the address manager.
+func (m *AddressManager) Run(ctx context.Context) error {
+	log.Debugf("Starting address manager.")
+	defer log.Debugf("Address manager stopped.")
+
+	runCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	m.runCtx = runCtx
+
+	// Communicate to the caller that the address manager has completed its
+	// initialization.
+	close(m.initChan)
+
+	<-runCtx.Done()
+
+	return nil
+}
+
+// NewAddress starts a new address creation flow.
+func (m *AddressManager) NewAddress(ctx context.Context) (*AddressParameters,
+	error) {
+
+	// Before we request a new address we ask the loop server to probe us.
+	// This ensures connectivity and obtains a L402 ID if we haven't gotten
+	// one earlier.
+	localPubKey := m.cfg.Lnd.NodePubkey
+	err := m.cfg.LoopClient.Server.Probe(ctx, 1, localPubKey, nil, nil)
```

Comment:
Do we need the probe? Since we're later also calling the server so L402 will be obtained there.

Reply:
The L402 is used server-side to evaluate if the client already has a static address or if a new one has to be created.

Reply:
Using @sputn1ck's fetch-method once available.

Reply:
Done!